src/compiler/checker.ts

```diff
@@ -12495,6 +12509,36 @@ namespace ts {
         }
 
         /**
+         * Checks that the return type provided is an instantiation of the global Promise<T> type

```

Comment:
@mhegazy you gave qualitied support [back in December](https://github.com/Microsoft/TypeScript/issues/5911#issuecomment-162979457) for treating return type annotations one way for ES3/5 (improve the error message), and another way for ES6+ (use `checkTypeAssignableTo`). Is this still the case?